examples/exampleutil/exampleutil.go

```diff
@@ -1,28 +1,98 @@
 package exampleutil
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"log"
+	"math/rand"
+	"net/http"
 	"strings"
 	"time"
 
-	"math/rand"
-
 	"github.com/bitcoin-sv/spv-wallet-go-client/config"
 )
 
-var ExampleConfig = config.NewDefaultConfig("http://localhost:3003")
+// NewDefaultConfig returns a new instance of the default example configuration.
+func NewDefaultConfig() config.Config {
+	return config.NewConfig()
+}
+
+// NewCustomConfig allows creating a custom configuration with optional parameters.
+func NewCustomConfig(addr string, timeout time.Duration) config.Config {
+	options := []config.Option{}
+
+	if addr != "" {
+		options = append(options, config.WithAddr(addr))
+	}
+
+	if timeout > 0 {
+		options = append(options, config.WithTimeout(timeout))
+	}
+
+	return config.NewConfig(options...)
+}
+
+// NewCustomConfig allows creating a custom configuration with optional parameters.
+func NewCustomConfigWithTransport(addr string, timeout time.Duration, transport *http.Transport) config.Config {
+	options := []config.Option{}
+
+	if addr != "" {
+		options = append(options, config.WithAddr(addr))
+	}
+
+	if timeout > 0 {
+		options = append(options, config.WithTimeout(timeout))
+	}
```

Comment:
IMHO, the usage part of the example should be clear and straightforward without too many custom functions for building the client or calling client method. Otherwise the user will be confused how to use it, or will copy paste those methods to every project - which is pointless - because he should use just a client not the example utility functions.